homeassistant/components/ecowitt/sensor.py

```diff
@@ -215,26 +221,35 @@
 }
 
 
+def get_sensor_key_by_name(name: str | None) -> str | None:
+    """Retrieve the sensor key from SENSOR_MAP by its name."""
+    for key, mapping in SENSOR_MAP.items():
+        if mapping.name == name:
+            _LOGGER.debug("get_sensor_key_by_name %s", key)
+            return key
+    return None
+
+
+def get_sensor_stype_by_name(name: str) -> EcoWittSensorTypes | None:
+    """Retrieve the sensor type (stype) from SENSOR_MAP by its name."""
+    for mapping in SENSOR_MAP.values():
+        if mapping.name == name:
+            _LOGGER.debug("get_sensor_stype_by_name %s", mapping.stype)
+            return mapping.stype
+    return None
```

Comment:
```suggestion
@dataclass
class SensorInfo:
    key: str
    stype: EcoWittSensorTypes

SENSOR_INFO_BY_NAME = { mapping.name: SensorInfo(key, mapping.stype) for key, mapping in SENSOR_MAP.items() }
```

So we avoid iterating over all values on each call